staging/src/github.com/openshift/openshift-controller-manager/pkg/build/controller/build/build_controller.go

```diff
@@ -366,6 +379,44 @@ func (bc *BuildController) setBuildDefaultProxy(spec *configv1.ProxySpec) {
 	bc.buildDefaults.DefaultProxy = spec
 }
 
+func (bc *BuildController) setBuildDefaultProxyFromProxyStatus(proxy *configv1.Proxy) {
+	bc.configLock.Lock()
+	defer bc.configLock.Unlock()
+	defaultProxy := bc.buildDefaults.DefaultProxy
+	if defaultProxy == nil {
+		defaultProxy = &configv1.ProxySpec{}
+		bc.buildDefaults.DefaultProxy = defaultProxy
+	}
+	if len(defaultProxy.HTTPProxy) == 0 {
+		defaultProxy.HTTPProxy = proxy.Status.HTTPProxy
+	}
```

Comment:
We shouldn't test against the current value - if the cluster admin updates the global proxy from a previous value, we'll miss the change.

Reply:
some more leftover from the prior form I did not clean up ... will address / thanks